File: app/src/main/java/org/breezyweather/common/ui/activities/AwakeUpdateActivity.java
package org.breezyweather.common.ui.activities;

import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import org.breezyweather.R;
import org.breezyweather.background.polling.PollingManager;

public class AwakeUpdateActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Toast.makeText(getApplicationContext(), R.string.refresh, Toast.LENGTH_SHORT).show();
        PollingManager.resetAllBackgroundTask(this, true);
        finish();
    }
}

File: app/src/main/java/org/breezyweather/common/ui/activities/AwakeUpdateActivity.kt
package org.breezyweather.common.ui.activities

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import org.breezyweather.R
import org.breezyweather.background.polling.PollingManager

class AwakeUpdateActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Toast.makeText(applicationContext, R.string.action_refresh, Toast.LENGTH_SHORT).show()
        PollingManager.resetAllBackgroundTask(this, true)
        finish()
    }
}
